step/src/main/java/com/google/sps/data/user/User.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data.user;
+
+import java.util.Collection;
+import java.util.HashSet;
+
+public class User {
+  
+  private String email;
+  private String name;
+  private Collection<User> matches;
```

Comment:
Did you consider the comment on the design doc.  Instead of storing the entire User, it would probably be better to assign each user a unique identifier. Then when you match the user, store the other user's unique identifier rather than the whole user.

Reply:
Ok, I can make this change. Since emails are unique, can I use them as unique identifiers or should I be generating an integer id for each user (since integers are cheaper to store than Strings)?

Reply:
Currently my code in master branch stores both emails and ID as unique identifiers. However, I use email to get access to users' data.

Reply:
I unique identifier is never any user input data, this is prone for errors. It should be unique generated long. The unique identifier should be something that will never change. Often times a user can change their email. @nganttran whenever possible, the unique id should be use to access user data rather than their email. Technically speaking, an email is part of the user data.

Reply:
You are right! It's better to set id as the unique identifier. Because we are using Google login as the only identification method. We don't allow to users to change the email. But I agree that the id is still better for the development in the future